File: app/src/main/java/android/academy/nytimes/network/NYTimesApi.java
package android.academy.nytimes.network;

import android.academy.nytimes.data.NewsItem;
import android.academy.nytimes.network.model.NYResponseRoot;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;
import retrofit2.http.Url;

interface NYTimesApi {

    @GET
    Call<NYResponseRoot> getNews(@Url String url, @Query("api-key") String apiKey);

}

File: app/src/main/java/android/academy/nytimes/network/NYTimesApi.kt
package android.academy.nytimes.network

import android.academy.nytimes.network.model.NYResponseRoot
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Url

interface NYTimesApi {

    @GET
    fun getNews(@Url url: String): Single<NYResponseRoot>

}
